components/datadog/apps/fakeintake/fargateFakeintakeService.go

```diff
@@ -79,6 +85,89 @@ func NewECSFargateInstance(e aws.Environment, option ...fakeintakeparams.Option)
 			return nil, err
 		}
 
+		ca, err := acmpca.NewCertificateAuthority(e.Ctx, namer.ResourceName("ca"), &acmpca.CertificateAuthorityArgs{
+			Type: pulumi.StringPtr("ROOT"),
+			CertificateAuthorityConfiguration: acmpca.CertificateAuthorityCertificateAuthorityConfigurationArgs{
+				KeyAlgorithm:     pulumi.String("RSA_4096"),
+				SigningAlgorithm: pulumi.String("SHA512WITHRSA"),
+				Subject: acmpca.CertificateAuthorityCertificateAuthorityConfigurationSubjectArgs{
+					CommonName: e.CommonNamer.DisplayName(64, pulumi.String("fakeintake CA")),
+				},
+			},
+			RevocationConfiguration: acmpca.CertificateAuthorityRevocationConfigurationArgs{
+				CrlConfiguration: acmpca.CertificateAuthorityRevocationConfigurationCrlConfigurationArgs{
+					Enabled: pulumi.BoolPtr(false),
+				},
+				OcspConfiguration: acmpca.CertificateAuthorityRevocationConfigurationOcspConfigurationArgs{
+					Enabled: pulumi.Bool(false),
+				},
+			},
+			PermanentDeletionTimeInDays: pulumi.IntPtr(7),
+		}, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		current, err := paws.GetPartition(e.Ctx, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+		caCert, err := acmpca.NewCertificate(e.Ctx, namer.ResourceName("ca-cert"), &acmpca.CertificateArgs{
+			CertificateAuthorityArn:   ca.Arn,
+			CertificateSigningRequest: ca.CertificateSigningRequest,
+			SigningAlgorithm:          pulumi.String("SHA512WITHRSA"),
+			TemplateArn:               pulumi.String(fmt.Sprintf("arn:%v:acm-pca:::template/RootCACertificate/V1", current.Partition)),
+			Validity: acmpca.CertificateValidityArgs{
+				Value: pulumi.String("10"),
+				Type:  pulumi.String("YEARS"),
+			},
+		}, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		if _, err = acmpca.NewCertificateAuthorityCertificate(e.Ctx, namer.ResourceName("ca-cert-cert"), &acmpca.CertificateAuthorityCertificateArgs{
+			CertificateAuthorityArn: ca.Arn,
+			Certificate:             caCert.Certificate,
+			CertificateChain:        caCert.CertificateChain,
+		}, opts...); err != nil {
+			return nil, err
+		}
+
+		domainName := alb.LoadBalancer.DnsName().ApplyT(func(dnsName string) *string {
+			if len(dnsName) <= 64 {
+				return &dnsName
+			}
+			dummyDNSName := "fakeintake.datadoghq.com"
+			return &dummyDNSName
+		}).(pulumi.StringPtrOutput)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
How does this work ? How is domain name used ? How dummy name works ?

Reply:
The `DomainName` field ends in the Subject CN field of the X509 certificate.
This was the legacy way of specifying for which domain a certificate is issued. That’s why it’s still mandatory.
But this field was suffering from some limitations like being unique and limited to 64 bytes.
That’s why it has now been superseded by the SAN field which allows to specify longer domains.

So, if the real DNS name fits inside the 64 bytes limit, let’s use it in the subject CN because it’s nicer.
But otherwise, what’s really important nowadays is to set it properly in the SAN field.

![image](https://github.com/DataDog/test-infra-definitions/assets/1437785/6cb803c1-8058-49cc-b615-c7ec9cdcae7e)